### PR TITLE
Add server option for seamless reconnect

### PR DESCRIPTION
**Pull request by BrennanConroy**
Also adds support in Newtonsoft.Json and MsgPack since the tests I was writing used Newtonsoft by default.

### CODE REVIEW

File: src/SignalR/clients/csharp/Client/test/FunctionalTests/Startup.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void Configure(IApplicationBuilder app)
 
         app.UseEndpoints(endpoints =>
         {
-            endpoints.MapHub<TestHub>("/default");
+            endpoints.MapHub<TestHub>("/default", o => o.AllowAcks = true);
```
Comment (mitchdenny) [inline src/SignalR/clients/csharp/Client/test/FunctionalTests/Startup.cs:89]: Should we expand this out to: ```AllowAcknowledgement[Something]```?

File: src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocolWorker.cs [hunk 2]
```diff
@@ -555,6 +575,20 @@ private static void WritePingMessage(ref MessagePackWriter writer)
...
+    {
+        writer.WriteArrayHeader(2);
+        writer.Write(HubProtocolConstants.AckMessageType);
+        writer.Write(message.SequenceId);
+    }
+
+    private static void WriteSequenceMessage(SequenceMessage message, ref MessagePackWriter writer)
+    {
+        writer.WriteArrayHeader(2);
+        writer.Write(HubProtocolConstants.SequenceMessageType);
```
Comment (mitchdenny) [inline src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocolWorker.cs:588]: I notice the docs for ```SequenceMessageType``` and ```AckMessageType``` on the ```HubProtocolConstants``` type are missing.

**Comment by mitchdenny** [comment]
LGTM, the only question I have is whether ```AllowAcks``` needs to go through API review.

**Comment by BrennanConroy** [comment]
> whether `AllowAcks` needs to go through API review

Technically yes, but it's unlikely this will be where the API exists after future changes, and I won't be around to do API review for a while. Planning on having an API review once more of the feature is completed and things are more final.

**Review by mitchdenny — APPROVED** [review]

**Comment by halter73** [comment]
Sorry you had to bother to merge this on vacation @BrennanConroy. I was planning to merge this for you today. I'm going to try to get #48577 merged today too.

**Pull request merged**